readme.md

```diff
@@ -0,0 +1,191 @@
+# remark for Visual Studio Code
+
+[![Build][build-badge]][build]
+[![Coverage][coverage-badge]][coverage]
+[![Downloads][downloads-badge]][downloads]
+[![Sponsors][sponsors-badge]][collective]
+[![Backers][backers-badge]][collective]
+[![Chat][chat-badge]][chat]
+
+Visual Studio Code extension to format and lint markdown files with remark.
+
+## Contents
+
+*   [What is this?](#what-is-this)
+*   [When should I use this?](#when-should-i-use-this)
+*   [Use](#use)
+    *   [Formatting](#formatting)
+*   [Plugins](#plugins)
+*   [Syntax](#syntax)
+*   [Compatibility](#compatibility)
+*   [Security](#security)
+*   [Contribute](#contribute)
+*   [License](#license)
+
+## What is this?
+
+This project is a Visual Studio Code (VS Code) extension that you can use in
+your editor to inspect and change markdown files.
+This extension is built around remark, which is a very popular ecosystem of
+plugins that work with markdown.
+You can choose from the 150+ plugins that already exist or make your own.
+
+See [remark][] for info on what the remark ecosystem is.
+
+## When should I use this?
+
+You can use this extension if you want to check (lint) and format markdown
+files from within your editor.
+
+To configure this extension, you define your preferred markdown style in a
+configuration file (`.remarkrc`, `.remarkrc.js`, etc. or in `package.json`).
+This file is picked up by `vscode-remark` and other tools (useful for
+contributors that don’t use VS Code).
+
+The configuration file is also used by [`remark-cli`][remark-cli], which is
+recommended to be used alongside `vscode-remark`, as an npm script and/or in
+CI, to enforce the markdown style.
+
+## Use
+
+To start using this for a project, [`remark-cli`][remark-cli] needs to be
+configured first.
+First, install `remark-cli` and a plugin or preset.
+Let’s use `remark-preset-lint-recommended` for our example.
+
+```sh
+npm install remark-cli remark-preset-lint-recommended
+```
+
+Now create a file named `.remarkrc.json` with the following content:
+
+```json
+{
+  "plugins": ["preset-lint-recommended"]
+}
+```
+
+Now let’s create a markdown file `readme.md` and add the following content:
+
+```markdown
+# My project
+
+This is a [broken link][]
+
+- A `*` should be used for list items
+- Also 4 spaces should be used for indentation
+
+[unused reference]: https://example.com
+```
+
+Now any syntax which doesn’t comply with `remark-preset-lint-recommended` will
+be reported using squiggly lines in your editor and in the `Problems` pane.
```

Comment:
````suggestion
This example enables VS Code users that have this extension installed to see
warnings about markdown style in their editor and the `Problems` pane and to
format markdown from a command.

Make sure that this extension is installed and then install the CLI and
plugins:

```sh
npm install remark-cli remark-preset-lint-recommended remark-preset-lint-consistent remark-toc
```

Then, add a `remarkConfig` to your `package.json` to configure remark:

```js
  /* … */
  "remarkConfig": {
    "settings": {
      "bullet": "*", // Use `*` for list item bullets (default)
      // See <https://github.com/remarkjs/remark/tree/main/packages/remark-stringify> for more options.
    },
    "plugins": [
      "remark-preset-lint-consistent", // Check that markdown is consistent.
      "remark-preset-lint-recommended", // Few recommended rules.
      [
        // Generate a table of contents in `## Contents`
        "remark-toc",
        {
          "heading": "contents"
        }
      ]
    ]
  },
  /* … */
```

> 👉 **Note**: you must remove the comments in the above examples when
> copy/pasting them, as comments are not supported in `package.json` files.

Now, you can open markdown files in your project, and you’ll see squiggly lines
and warnings in the `Problems` pane.
Here’s an example that should produce problems you can use to verify:

```markdown
1) Hello, _Jupiter_ and *Neptune*!
```
````

Reply:
- This is based on the main example in `remark-cli` and some examples in `remark-lint`.
- As we assume we’re in an npm package, we can place config in `package.json` instead of a separate file
- This shows more examples of plugins/presets, including how to pass options
- Q: is `remark-cli` needed here?

Reply:
`remark-parse` and `remark-stringify` are needed, which are pulled in by `remark-cli`. I’d just suggest to install `remark-cli`.